server/src/main/java/de/uftos/services/ConstraintInstanceService.java

```diff
@@ -123,16 +128,21 @@ public ConstraintInstance create(String signatureId, ConstraintInstanceRequestDt
    *
    * @param signatureId the ID of the signature for the instances.
    * @param pageable    contains the parameters for the page.
-   * @return the page of the entries fitting the parameters.
+   * @return the entries fitting the parameters.
    */
-  public Page<ConstraintInstance> get(String signatureId, Pageable pageable,
-                                      Optional<String> argument) {
+  public ConstraintInstanceResponseDto get(String signatureId, Pageable pageable,
+                                           Optional<String> argument) {
+    // TODO search for "argument" in all resources relations and get all IDs of the resources
+    //  that fulfill the filter, then merge all the IDs with the IDs of the arguments of the
+    //  instances of this specific signature. THEN return the instances that own these arguments.
+    //  GOOD LUCK!
     Specification<ConstraintInstance> specification = new SpecificationBuilder<ConstraintInstance>()
-        .andJoinEquals(signatureId, "signature", "name")
-        .optionalAndJoinLikeIgnoreCase(argument, "arguments", "value")
         .build();
 
-    return this.repository.findAll(specification, pageable);
+    Page<ConstraintInstance> constraintInstances = this.repository.findAll(specification, pageable);
+    List<ConstraintArgumentDisplayName> displayNames =
+        processConstraintInstances(constraintInstances.getContent());
```

Comment:
`process` is like extremely generic. Could we be more specific in the naming here?

Reply:
done